app/assets/javascripts/darkswarm/services/customer.js.coffee

```diff
@@ -13,7 +13,17 @@ angular.module("Darkswarm").factory 'Customer', ($resource, Messages) ->
 
   Customer.prototype.update = ->
     @$update().then (response) =>
-      Messages.success(t('js.changes_saved'))
+      if response.gateway_setup_intent_client_secret && $injector.has('stripeObject')
+        stripe = $injector.get('stripeObject')
+        stripe.confirmCardSetup(response.gateway_setup_intent_client_secret).then (result) =>
```

Comment:
The way it's working now (which is definitely not the only way it could work) is that we create the SetupIntent in the controller only if the `allow_charges` param is true. I considered making a separate endpoint in the controller (something like `create_setup_intent`) and kept coming back to this way of doing it. There are cases where `confirmCardSetup` just succeeds without the user having to do anything (but typing that out loud, makes me realize that's not really an argument against what you're suggesting). 

It's also partly me not being very fluent with Angular to the point where I could modify the existing code so that checking the box didn't set `allow_charges` to `true` (or, put another way, my preference was to keep what was already simple and working on the Angular side). 

I agree though that ideally we wouldn't set the field until it's actually confirmed by Stripe.

Reply:
I see. Maybe a checkbox is not the ideal solution here but I'm happy to keep it. Visually, I could imagine that the user ticks the box and it becomes read-only until the request is done. Then it's either still ticked or it gets unticked again and an error message is displayed. And that would all work with your current logic. All good, keep it.

Reply:
I did do something similar to what you suggest (the box becoming read-only) in this commit in part 2: https://github.com/andrewpbrett/openfoodnetwork/pull/3/commits/681d2c91062c9606b757374cf9bb6266d4db7d90#diff-4284e5c68dddf3754015403a55553605a0c86d42e9b9603f24dde7101adf649cR16